### PR TITLE
Properly estimate ibc gas fees

### PR DESCRIPTION
**Pull request by grod220**
Closes https://github.com/penumbra-zone/web/issues/1049

Before we were hardcoding a standard fee. This doesn't work generally across chains. Now we are doing tx simulations to get these estimates.

### CODE REVIEW

File: apps/minifront/src/state/ibc-in.ts [hunk 1]
```diff
@@ -109,6 +111,28 @@ const getCompatibleBech32 = (chainName: string, address: Address): string => {
...
+  client,
+  signerAddress,
+  message,
+}: {
+  chainId: string;
+  client: SigningStargateClient;
+  signerAddress: string;
+  message: EncodeObject;
+}) => {
+  const feeToken = chains.find(({ chain_id }) => chain_id === chainId)?.fees?.fee_tokens[0];
```
Comment (Valentine1898) [inline apps/minifront/src/state/ibc-in.ts:125]: can there be several `feeTokens`? can there be a case when among 2 `feeTokens` we choose the one which is not on the user's balance?

File: apps/minifront/src/state/ibc-in.ts [hunk 2]
```diff
@@ -109,6 +111,28 @@ const getCompatibleBech32 = (chainName: string, address: Address): string => {
...
+  const avgGasPrice = feeToken?.average_gas_price;
```
Comment (Valentine1898) [inline apps/minifront/src/state/ibc-in.ts:126]: Is this the accepted approach? If gas prices suddenly rise, all transactions we initiate will fail?

**Review by Valentine1898 — APPROVED** [review]
Looks good to me, just a few clarifying questions

File: apps/minifront/src/state/ibc-in.ts (hunk 1 above)
Comment (grod220) [reply apps/minifront/src/state/ibc-in.ts:125]: In theory yes. However, in practice, the cosmos chain registry for the chains we are connected to all have one. We may need to revisit this later if we connect to a chain and need to build in logic for choosing a gas token.

File: apps/minifront/src/state/ibc-in.ts (hunk 2 above)
Comment (grod220) [reply apps/minifront/src/state/ibc-in.ts:126]: This is what is employed by cosmos kit (using the average price published in the chain registry). Nearly every example of using cosmjs hardcodes this value. However, I share your sentiment. It feels non-flexible. Going to follow up with cosmology team and get their feedback on this. Will merge this for now though.

**Pull request merged**